DTSPO-17278 further improve shuttering

ieuanb74 opened the pull request:
### Jira link (if applicable)
https://tools.hmcts.net/jira/browse/DTSPO-17278


### Change description ###
- Removed zone level tagging and associated logic to reduce complexity
- Applied tagging to individual records
- Renamed area to platform 
- Removed `zone_area` var and references to it 

### Checklist

<!-- Check each box by removing the space and adding an x, e.g. [x] -->

- [x] commit messages are meaningful and follow good commit message guidelines
- [x] README and other documentation has been updated / added (if needed)
- [x] tests have been updated / new tests has been added (if needed)
- [ ] Does this PR introduce a breaking change

hmcts-platform-operations commented:
## Plan Result (Sandbox)




<pre><code>No changes. Your infrastructure matches the configuration.</code></pre>









<!-- github-comment: {"Command":"plan","PRNumber":1786,"Program":"tfcmt","SHA1":"","Target":"Sandbox","Vars":{}} -->

hmcts-platform-operations commented:
## Plan Result (Demo)




<pre><code>No changes. Your infrastructure matches the configuration.</code></pre>









<!-- github-comment: {"Command":"plan","PRNumber":1786,"Program":"tfcmt","SHA1":"","Target":"Demo","Vars":{}} -->

hmcts-platform-operations commented:
## Plan Result (Dev)




<pre><code>No changes. Your infrastructure matches the configuration.</code></pre>









<!-- github-comment: {"Command":"plan","PRNumber":1786,"Program":"tfcmt","SHA1":"","Target":"Dev","Vars":{}} -->

hmcts-platform-operations commented:
## Plan Result (Ithc)




<pre><code>No changes. Your infrastructure matches the configuration.</code></pre>









<!-- github-comment: {"Command":"plan","PRNumber":1786,"Program":"tfcmt","SHA1":"","Target":"Ithc","Vars":{}} -->

hmcts-platform-operations commented:
## Plan Result (nle)




<pre><code>No changes. Your infrastructure matches the configuration.</code></pre>









<!-- github-comment: {"Command":"plan","PRNumber":1786,"Program":"tfcmt","SHA1":"","Target":"nle","Vars":{}} -->

hmcts-platform-operations commented:
## Plan Result (Staging)




<pre><code>No changes. Your infrastructure matches the configuration.</code></pre>









<!-- github-comment: {"Command":"plan","PRNumber":1786,"Program":"tfcmt","SHA1":"","Target":"Staging","Vars":{}} -->

hmcts-platform-operations commented:
## Plan Result (Prod)




<pre><code>Plan: 0 to add, 4 to change, 0 to destroy.</code></pre>

* Update
  * module.ctsc-hmcts.azurerm_dns_cname_record.this[&#34;_B5AD23A3B52771815F77A4C7A85E6345.wfmapp&#34;]
  * module.ctsc-hmcts.azurerm_dns_cname_record.this[&#34;_E0BDCD3E999D84AEC133A95F71EE1F18.wfmapp.preprod&#34;]
  * module.ctsc-hmcts.azurerm_dns_cname_record.this[&#34;_E0BDCD3E999D84AEC133A95F71EE1F18.wfmapp.training&#34;]
  * module.prod-platform.azurerm_dns_cname_record.this[&#34;_AC285A78514B19D657B18C29B98F4C7F.vh-admin-web&#34;]




<details><summary>Change Result (Click me)</summary>

```hcl

  # module.ctsc-hmcts.azurerm_dns_cname_record.this["_B5AD23A3B52771815F77A4C7A85E6345.wfmapp"] will be updated in-place
  ~ resource "azurerm_dns_cname_record" "this" {
        id                  = "/subscriptions/ed302caf-ec27-4c64-a05e-85731c3ce90e/resourceGroups/reformmgmtrg/providers/Microsoft.Network/dnsZones/ctsc.hmcts.net/CNAME/_b5ad23a3b52771815f77a4c7a85e6345.wfmapp"
        name                = "_b5ad23a3b52771815f77a4c7a85e6345.wfmapp"
      ~ record              = "DBB483EBD37876263819222CEE3CD694.ECECD324B9EF785188CC31B79E103A02.9f5f53f37396a3c5c13d.sectigo.com." -> "DBB483EBD37876263819222CEE3CD694.ECECD324B9EF785188CC31B79E103A02.c11436178b6f4613b928.sectigo.com."
        tags                = {}
        # (5 unchanged attributes hidden)
    }

  # module.ctsc-hmcts.azurerm_dns_cname_record.this["_E0BDCD3E999D84AEC133A95F71EE1F18.wfmapp.preprod"] will be updated in-place
  ~ resource "azurerm_dns_cname_record" "this" {
        id                  = "/subscriptions/ed302caf-ec27-4c64-a05e-85731c3ce90e/resourceGroups/reformmgmtrg/providers/Microsoft.Network/dnsZones/ctsc.hmcts.net/CNAME/_e0bdcd3e999d84aec133a95f71ee1f18.wfmapp.preprod"
        name                = "_e0bdcd3e999d84aec133a95f71ee1f18.wfmapp.preprod"
      ~ record              = "603D81996C47863EDFE41A6ED2EF3224.A1363BE712CA02F75C3C5259EC7D4680.586ee5cb5f17541372cd.sectigo.com." -> "603D81996C47863EDFE41A6ED2EF3224.A1363BE712CA02F75C3C5259EC7D4680.9b1211e0578d0fbcf7ab.sectigo.com."
        tags                = {}
        # (5 unchanged attributes hidden)
    }

  # module.ctsc-hmcts.azurerm_dns_cname_record.this["_E0BDCD3E999D84AEC133A95F71EE1F18.wfmapp.training"] will be updated in-place
  ~ resource "azurerm_dns_cname_record" "this" {
        id                  = "/subscriptions/ed302caf-ec27-4c64-a05e-85731c3ce90e/resourceGroups/reformmgmtrg/providers/Microsoft.Network/dnsZones/ctsc.hmcts.net/CNAME/_e0bdcd3e999d84aec133a95f71ee1f18.wfmapp.training"
        name                = "_e0bdcd3e999d84aec133a95f71ee1f18.wfmapp.training"
      ~ record              = "603D81996C47863EDFE41A6ED2EF3224.A1363BE712CA02F75C3C5259EC7D4680.586ee5cb5f17541372cd.sectigo.com." -> "603D81996C47863EDFE41A6ED2EF3224.A1363BE712CA02F75C3C5259EC7D4680.9b1211e0578d0fbcf7ab.sectigo.com."
        tags                = {}
        # (5 unchanged attributes hidden)
    }

  # module.prod-platform.azurerm_dns_cname_record.this["_AC285A78514B19D657B18C29B98F4C7F.vh-admin-web"] will be updated in-place
  ~ resource "azurerm_dns_cname_record" "this" {
        id                  = "/subscriptions/ed302caf-ec27-4c64-a05e-85731c3ce90e/resourceGroups/reformmgmtrg/providers/Microsoft.Network/dnsZones/prod.platform.hmcts.net/CNAME/_ac285a78514b19d657b18c29b98f4c7f.vh-admin-web"
        name                = "_ac285a78514b19d657b18c29b98f4c7f.vh-admin-web"
      ~ record              = "032DDB9A61BB0588601DD6D4C005A812.302202E1FB30DE99CD976C202BDB71AC.fbb25c1fc3c9ad12a80d.sectigo.com." -> "032DDB9A61BB0588601DD6D4C005A812.302202E1FB30DE99CD976C202BDB71AC.d7703fe5d42a59a77748.sectigo.com."
        tags                = {}
        # (5 unchanged attributes hidden)
    }

Plan: 0 to add, 4 to change, 0 to destroy.
```

</details>





<!-- github-comment: {"Command":"plan","PRNumber":1786,"Program":"tfcmt","SHA1":"","Target":"Prod","Vars":{}} -->

endakelly commented:
Do we need some info in the main readme to say that teams should ensure platform is defined or will it assume a default value?

ieuanb74 commented:
> Do we need some info in the main readme to say that teams should ensure platform is defined or will it assume a default value?

Yep good shout. Have also added this to HMCTS way under the section about configuring public dns

endakelly commented:
> > Do we need some info in the main readme to say that teams should ensure platform is defined or will it assume a default value?
> 
> Yep good shout. Have also added this to HMCTS way under the section about configuring public dns
> 
> Actually might not be required as already link to HMCTS way and the example in this repo for more info on tagging - [hmcts/azure-public-dns#shuttering](https://github.com/hmcts/azure-public-dns?tab=readme-ov-file#shuttering)

Yeah, I think the overall guidance is fine, I just don't see an explicit example showing `platform` or `area` being set. Maybe just add a link to a record for cft and sds where the value has been set so people know what it should look like.

ieuanb74 commented:
> > > Do we need some info in the main readme to say that teams should ensure platform is defined or will it assume a default value?
> > 
> > 
> > Yep good shout. Have also added this to HMCTS way under the section about configuring public dns
> > Actually might not be required as already link to HMCTS way and the example in this repo for more info on tagging - [hmcts/azure-public-dns#shuttering](https://github.com/hmcts/azure-public-dns?tab=readme-ov-file#shuttering)
> 
> Yeah, I think the overall guidance is fine, I just don't see an explicit example showing `platform` or `area` being set. Maybe just add a link to a record for cft and sds where the value has been set so people know what it should look like.

I've updated this with a link to example

ieuanb74 commented on modules/azure-public-dns/a_record.tf:
```diff
@@ -1,18 +1,18 @@
   # Extract shutter_all_a value if available or false
   # If global shutter is true, set to true otherwise set to value of shutter_all_a
-  shutter_all_a = local.shutter_all == true ? true : (var.zone_area == "cft" && local.shutter_all_cft == true) ? true : (var.zone_area == "sds" && local.shutter_all_sds == true) ? true : lookup(yamldecode(var.shutter_config), "shutter_all_a", false)
+  shutter_all_a = local.shutter_all == true ? true : lookup(yamldecode(var.shutter_config), "shutter_all_a", false)
```
No longer required as have removed this in favour of tagging at the record level only. Same experience for user and makes the logic easier to read
